### ISSUE TITLE
Clusters do not update after moving Markers?

### ISSUE THREAD

**Issue by aalbericio** [Leaflet/Leaflet.markercluster#649]
Hello,

This plugin works fine when adding Markers in the LayerClusterGroup:

**var newMarker = L.marker(...);**

**newMarker.addTo(myCluster);**

But it does not seem to work when updating some already created marker's position:

**foundMarker.setLatLng(...);**

The marker changes position but that change does not update the clusters in the LayerClusterGroup...

Is there anything I can do to **refresh** the LayerClusterGroup everytime any of its markers changes its position?

Thanks in advance.

**Comment by aalbericio** [comment]
Hello again,

I have found a workaround that is far from being optimum (I think):

`function rebuildClusters() {
		myCluster.clearLayers();
		
		for (var key in markers) {
        	      markers[key].addTo(myCluster);
		}
	}`

I dont like the solution but it just works. 

What do you think? Is there any way/method I can call on the cluster that performs this re-calculation with optimum performance?

Thank you in advance.

**Comment by ghybs** [comment]
Hi,

Searching in this repo issues for "moving marker" should give you plenty threads, e.g.:
- #57 Allow for moving markers without removing and re-adding them
- #122 Updating marker position
- #266 Updating MarkerCluster Position
- #304 A KD-Tree version (=> pruneCluster plugin)
- #456 Cluster does not "fresh" when markers are moving
- #459 refresh clusters

TL;DR
  1. Remove from MCG the markers that are going to move (recommend using `removeLayers`).
  2. Move the markers.
  3. Add them back into MCG (recommend using `addLayers`).

If you use Leaflet 1.0.0 and Leaflet.markercluster 1.0.0, this should be automatic.

Hope this helps.

**Comment by aalbericio** [comment]
Thank you, Danzel.

That piece code fixed the problem. Anyway, if a marker has state (it is an HTML with states), removing it and readding it makes it lose state and, of course, it is not a best practice to remove/add a marker just because it has changed its location. That operation can be "expensive".

What do you think?

Cheers

**Comment by danzel** [comment]
I think you've misunderstood.
If you are using the 1.0.0 versions then calling ```setLatLng``` on your marker works correctly with MCG, you don't need to add/remove.
